### ISSUE TITLE
V7 resolvers issue - field array errors

### ISSUE THREAD

**Issue by wdfinch** [react-hook-form/react-hook-form#4321]
Right now it seems that there is no way to get string errors via schema validation when the error is not related to a specific field. So in my case, I want to validate that string inputs in a field array are unique. This works fine, however, I would like to display a string error message to the user stating what the error is. Right now it seems that there is no way to access this message. Therefore I'm wondering if it would be possible to add error messages to `formState.errors` for entire field arrays in addition to field array inputs. @jorisre maybe you would have some good input here. 

https://codesandbox.io/s/react-hook-form-usefieldarray-template-forked-m9lgw?file=/src/index.js

Notice that in the code sandbox that the `formState.isValid` is false as it should be but `formState.errors` is empty.

Sorry for the edited post. I accidentally posted this as soon as I opened it.

**Comment by jorisre** [comment]
Hi @wdfinch 👋🏻 

I think we can't add error messages to `formState.errors` because resolver doesn't know about errors. It just formats in React Hook Form error format.

Below, the Joi original error, as you can see, the path is `names[1]`

```json
[
   {
      "message":"\"names[1]\" contains a duplicate value",
      "path":[
         "names",
         1
      ],
      "type":"array.unique",
      "context":{
         "pos":1,
         "value":{
            "name":"will"
         },
         "dupePos":0,
         "dupeValue":{
            "name":"will"
         },
         "path":"name",
         "label":"names[1]",
         "key":1
      }
   }
]
```

I've got a possible workaround, using `custom` Joi validation

Is that you expect ?

```json
{
   "names":{
      "message":"Duplicate name 'will'",
      "type":"custom"
   }
}
```
 👉🏻 https://codesandbox.io/s/react-hook-form-usefieldarray-template-forked-l2g4d

Joi helpers's documentation to create better error
https://joi.dev/api/?v=17.4.0#validation-helpers

**Comment by wdfinch** [comment]
Thanks for the reply. Yes, so I would like to receive something like what you posted. I can easily modify the error message and adjust the validation check from joi. However, in the sandbox you posted, `formState.errors `is still an empty object. Is there any way we could get 

```
{
   "names":{
      "message":"Duplicate name 'will'",
      "type":"custom"
   }
}
```

into formState.errors?

**Comment by wdfinch** [comment]
Sorry, you mentioned that `formState.errors` can't be modified. Is there any way we could access this then without manually running the joi validation with our `formState` and then passing the result around in some manner? Could resolvers add some option or helper to access error messages that don't make it to `formState.errors`?

**Comment by bluebill1049** [comment]
If I follow you correctly. @wdfinch I think you will have to write your own custom resolver, for an error that is not part of the form/fields.

- isValid will return again the schema itself
- errors message will be associated with all your inputs. 

external errors will need to be mapped at your app-level with a custom resolver.

<img width="1106" alt="Screen Shot 2021-03-02 at 12 28 14 pm" src="https://user-images.githubusercontent.com/10513364/109582735-c6562080-7b52-11eb-9850-8f186225df3a.png">

**Comment by wdfinch** [comment]
I see. It would be nice if we could make this easier to do as it's not that uncommon of a use case. For example, field array must have at least one element is something I use on every field array input.

Would you be open to possibly having a `schemeErrors` key in `formState.errors` or some other place that we could access this in `formState`?

**Comment by bluebill1049** [comment]
> Would you be open to possibly having a schemeErrors key in formState.errors or some other place that we could access this in formState?

not at this stage, until demand rise we can look into that.

**Comment by wdfinch** [comment]
Ok, then I guess we close this for now unless you guys have any suggestions.

**Comment by bluebill1049** [comment]
ok, I Will be keep an eye on this issue or similar issue before we plan something for this use case.

**Comment by justjuank** [comment]
I came across the same requirement trying to validate duplicate values in an array, I can add the validation using Yup directly to the array field with a custom test, but there is no way to retrieve the error message to display it.

The only workaround I found for now was to add the validation directly to the property of the array item (assuming it's an array of objects) that I want to check for duplicates and validate it against the entire array passed in the context of the Yup `test` method.

I agree it would be nice to have errors not related to a specific field.
